File: it.stefano.bls.domain/lib/src/main/java/button/concrete/ButtonDevice.java
package button.concrete;

import interf.IButtonDevice;
import observable.GenericObservable;

public class ButtonDevice extends GenericObservable implements IButtonDevice {

	private boolean state = false;
	
	protected void setState(boolean aState) {
		state = aState;
		updateObservers();
	}

	private void updateObservers() {
		String state = this.state ? "pressed" : "released";
		updateObservers(state);
	}

}

File: it.stefano.bls.arduino/libraries/Button/ButtonDevice.cpp
#include "ButtonDevice.h"

void ButtonDevice::setState(bool state) {
  this->state = state;
  if(state) {
   setData(String("pressed"));
   //Serial.println("Button | set state pressed");
  } else {
    setData(String("released"));
  }
}
bool ButtonDevice::getState() {
  return this->state;
}